### PR TITLE
Solves #22 and greatly improves the project documentation

### PR DESCRIPTION
**Pull request by Sharcoux**
Add types for use with intellisense, typescript, etc, and improve project documentation

Solve https://github.com/derhuerst/svg-radar-chart/issues/22

### CODE REVIEW

**Review by derhuerst — COMMENTED** [review]
Can you keep the formatting (no trailing semicolons, indentation, etc.) as before? The current changeset makes reviewing (and blaming in the future) very hard.

**Comment by Sharcoux** [comment]
You should probably define those in a linter config, this way the formatting will always follow the same rules.

**Comment by derhuerst** [comment]
> You should probably define those in a linter config, this way the formatting will always follow the same rules.

I see your point! But this creates more maintenance work, and I would like to rather make it less. See also https://github.com/public-transport/hafas-client/pull/305#issuecomment-1837485331 .

**Comment by Sharcoux** [comment]
This is the opposite. The linter will automatically ensure that the code style is the same throughout the project. I don't like prettier either, but you can perfectly lint code with eslint, which is already installed in the project. If you check my last change, it implements some of your requirements.

File: index.js [hunk 1]
```diff
@@ -49,73 +132,99 @@ const shape = (columns, opt) => (data, i) => {
...
 		transform: `translate(${delta},${delta})`
 	}, groups)
 }
 
+// Export the radar chart rendering function
 export {
 	renderRadarChart as radar,
 }
+
+export * from './smoothing.js'
```
Comment (derhuerst) [inline index.js]: Please revert this change. I want to keep the smoothing part completely separate to allow smaller bundle sizes (when no tree-shaking-like processing is in place).

**Comment by derhuerst** [comment]
Adding a linter creates long-term friction for me, this is why I don't want to have such a setup. (I can see that using it is worth it in projects with multiple active maintainers and/or many contributors.)

Thanks for adjusting your PR accordingly.

File: index.js (hunk 1 above)
Comment (Sharcoux) [reply index.js]: Done

**Comment by Sharcoux** [comment]
> Adding a linter creates long-term friction for me, this is why I don't want to have such a setup. (I can see that using it is worth it in projects with multiple active maintainers and/or many contributors.)
> 
> Thanks for adjusting your PR accordingly.

You already have a linter. eslint is a linter - there is a hint in the name :wink:

**Review by derhuerst — APPROVED** [review]
